### DISCUSSION TITLE
OpenAPI data models unexpectedly split into input / output variations

### DISCUSSION THREAD

**Discussion by edinpeter** [fastapi/fastapi#11371] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel
from enum import Enum

app = FastAPI()


class LogLevel(str, Enum):
    DEBUG = "debug"
    INFO = "info"
    WARNING = "warning"
    ERROR = "error"
    CRITICAL = "critical"

class Log(BaseModel):
    log_level: LogLevel


class Logs(BaseModel):
    logs: list[Log]

@app.get("/logs", response_model=Logs)
def get_logs():
    pass
    return None

@app.put("/logs",)
def update_logs(body: Logs):
    pass
    return None

```

### Description

The fastapi documents explain this functionality here: https://fastapi.tiangolo.com/how-to/separate-openapi-schemas/

The only cause of this behavior they indicate is that when optional values are present with defaults, the openapi models will get split into *Input and *Output. 

However, with this minimum example, the data models do not have any nullable fields, nor defaults, and appear identical but were split anyway: ![Screen Shot 2024-03-29 at 3 26 42 PM](https://github.com/tiangolo/fastapi/assets/8761244/997d692c-33c3-4ab9-a568-5a7b6f35a92c)

Is there something I'm missing in what causes this behavior? Removing the `response_model=...` statement reverts to what I would expect to happen, with a single model- Logs in the OpenAPI spec.

### Operating System

macOS

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

3.11

**Comment by a-schulz** [comment]
@edinpeter 
Might be a bit late but check the documentation:
https://fastapi.tiangolo.com/how-to/separate-openapi-schemas/#model-for-input-and-output-in-docs
